content/Objects/Objects.q.md

```diff
@@ -0,0 +1,1040 @@
+
+Execute the program. What is its output?
+
+ - It prints `eat`.
+
+ - It prints `Diana`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+
+/// type=SS, answer=[5]
+
+Which of the following is an object?
+
+ - `new`
+
+ - `eat()`
+
+ - `$name`
+
+ - `Person`
+
+ - `$personObject`
+
+
+/// type=SS, answer=[3]
+
+Which of the following is a property?
+
+ - `new`
+
+ - `eat()`
+
+ - `$name`
+
+ - `Person`
+
+ - `$personObject`
+
+
+/// type=SS, answer=[2]
+
+Which of the following is a method?
+
+ - `new`
+
+ - `eat()`
+
+ - `$name`
+
+ - `Person`
+
+ - `$personObject`
+
+
+/// type=SS, answer=[1]
+
+On line 13, what does `$personObject->eat()` do?
+
+ - It calls the `eat()` method of `$personObject`.
+
+ - It defines the `eat()` method of `$personObject`.
+
+ - It creates the `eat()` method of `$personObject`.
+
+ - It declares the `eat()` method of `$personObject`.
+
+ - It displays the `eat()` method of `$personObject`.
+
+
+/// type=SS, answer=[5]
+
+In the statement `$personObject->eat();` on line 13, what does the object operator `->` do?
+
+ - It sets the `eat()` method of `$personObject`.
+
+ - It creates the `eat()` method of `$personObject`.
+
+ - It defines the `eat()` method of `$personObject`.
+
+ - It declares the `eat()` method of `$personObject`.
+
+ - It accesses the `eat()` method of `$personObject`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+
+    $personObject = new Person();
+    $personObject->name = "Charles";
+    echo $personObject->name;
+?>
+```
+/// type=SS, answer=[3]
+
+Execute the program. What is its output?
+
+ - It prints `name`.
+
+ - It prints `Diana`.
+
+ - It prints `Charles`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+
+/// type=SS, answer=[2]
+
+Which statement best describes the code on line 13?
+
+ - It adds the value `Charles` of the `name` property to `$personObject`.
+
+ - It sets the value of the `name` property of `$personObject` to `Charles`.
+
+ - It returns the value `Charles` of the `name` property of `$personObject`.
+
+ - It displays the value `Charles` of the `name` property of `$personObject`.
+
+ - It removes the value `Charles` of the `name` property of `$personObject`.
+
+
+/// type=SS, answer=[3]
+
+On lines 12, 13, and 14, replace the `$personObject` object with `$john`. Execute the program. What is its output?
+
+ - It prints `name`.
+
+ - It prints `Diana`.
+
+ - It prints `Charles`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+
+    $john = new Person();
```

Comment:
Naming the variable $john seems a little confusing and arbitrary when his name is Charles. Maybe $person

Reply:
My mistakes. I will do some changes on this. Thank you, T for pointing this out.